### PR TITLE
Supplies - Add randomized box functionality

### PR DESCRIPTION
**Pull request by JoramD0**
**When merged this pull request will:**
- title

### CODE REVIEW

File: addons/supplies/XEH_postInitClient.sqf [hunk 1]
```diff
@@ -10,3 +10,38 @@ if (!hasInterface) exitWith {};
...
+    "tacgt_Lootbox_SMG",
+    "tacgt_Lootbox_Rifle",
+    "tacgt_Lootbox_GL",
+    "tacgt_Lootbox_LMG",
+    "tacgt_Lootbox_Marksman"
+];
+
+{
+    [
+        _x,
```
Comment (jonpas) [inline addons/supplies/XEH_postInitClient.sqf:26]: Create a base class here in Mods which GTArmory can then inherit from. You can only add it to the base class and context menu will apply for inherited items as well. I recommend `GVAR(randomizedBox)` or anything over "lootbox".

Also change the config entry to `GVAR(possibleItems)` or something similar.

Comment (JoramD0) [reply addons/supplies/XEH_postInitClient.sqf:26]: @TyroneMF This sounds like a you thing

File: addons/supplies/CfgWeapons.hpp [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class CfgWeapons {
+    class CBA_MiscItem;
+    class CBA_MiscItem_ItemInfo;
+    class GVAR(replacementBox): CBA_MiscItem {
```
Comment (JoramD0) [inline addons/supplies/CfgWeapons.hpp:4]: It might be better to name this more specific to the random chance mechanic.

Comment (TyroneMF) [reply addons/supplies/CfgWeapons.hpp:4]: ..Pick a name, let me know.

Comment (jonpas) [reply addons/supplies/CfgWeapons.hpp:4]: `GVAR(randomizedBox)` :P

File: addons/supplies/XEH_postInitClient.sqf [hunk 3]
```diff
@@ -10,3 +10,27 @@ if (!hasInterface) exitWith {};
 }] call CBA_fnc_addEventHandler;
 
 [QGVAR(loadTraits), ace_player] call CBA_fnc_serverEvent;
+
+// Lootbox functionality
+[
+    QGVAR(randomizedBox),
+    "CONTAINER",
+    LLSTRING(OpenLootbox),
```
Comment (jonpas) [inline addons/supplies/XEH_postInitClient.sqf:18]:
```suggestion
    LLSTRING(OpenRandomBox),
```

**Pull request merged**